trio/_core/_run.py

```diff
@@ -2533,6 +2678,7 @@ async def checkpoint() -> None:
     await cancel_shielded_checkpoint()
     task = current_task()
     task._cancel_points += 1
+    assert task._cancel_status is not None, task
```

Comment:
`checkpoint` and `checkpoint_if_cancelled` are very hot perf-wise. I think it might be better to use a type-ignore than an assertion here. In practice the only way to get `Task._cancel_status is None` is in the init task outside the outermost nursery, and there's no user code that runs there. Typing `Task` as if `_cancel_status` can never be None is also reasonable IMO.

Reply:
Yeah it might make more sense to make it non-optional, then type-ignore the places where it is `None`.